modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalSpecTests.java

```diff
@@ -70,23 +72,17 @@ public static void afterClass() throws Exception {
         searchRequestParsers = null;
     }
 
-    public void testRoundtripping() throws IOException {
-        List<String> indices = new ArrayList<>();
-        int size = randomIntBetween(0, 20);
+    private static <T> List<T> randomList(Supplier<T> randomSupplier) {
+        List<T> result = new ArrayList<>();
+        int size = randomIntBetween(1, 20);
         for (int i = 0; i < size; i++) {
-            indices.add(randomAsciiOfLengthBetween(0, 50));
+            result.add(randomSupplier.get());
         }
+        return result;
+    }
 
-        List<String> types = new ArrayList<>();
-        size = randomIntBetween(0, 20);
-        for (int i = 0; i < size; i++) {
-            types.add(randomAsciiOfLengthBetween(0, 50));
-        }
-        List<RatedRequest> specs = new ArrayList<>();
-        size = randomIntBetween(1, 2); // TODO I guess requests with no query spec should be rejected...
-        for (int i = 0; i < size; i++) {
-            specs.add(RatedRequestsTests.createTestItem(indices, types));
-        }
+    private RankEvalSpec createTestItem(List<String> indices, List<String> types) throws IOException {
+        List<RatedRequest> ratedRequests = randomList(() -> RatedRequestsTests.createTestItem(indices, types));
```

Comment:
I forgot to ask again about the necessity for passing in the indices and types for random item creation here. Can this be included into `createTestItem` in both RatedRequestsTests and thereby also removing to generate those parameters outside of `createTestItem` here?